plugins/ocm-backend/src/service/router.ts

```diff
@@ -101,11 +107,24 @@ export async function createRouter(
     const allClusters = await Promise.all(
       Object.values(clients).map(async c => {
         const mcs = await listManagedClusters(c.client);
-
-        return mcs.items.map(mc => ({
-          name: translateOCMToResource(mc.metadata!.name!, c.hubResourceName),
-          status: parseClusterStatus(mc),
-        }));
+        const mcis = await listManagedClusterInfos(c.client);
+
+        return mcs.items.map(
+          (mc, index) =>
+            ({
+              name: translateOCMToResource(
+                mc.metadata!.name!,
+                c.hubResourceName,
+              ),
+              status: parseClusterStatus(mc),
+              platform: getClaim(mc, 'platform.open-cluster-management.io'),
+              openshiftVersion:
+                mc.metadata!.labels?.openshiftVersion ||
+                getClaim(mc, 'version.openshift.io'),
+              nodes: parseNodeStatus(mcis.items[index]),
+              ...parseUpdateInfo(mcis.items[index]),
+            } as ClusterOverview),
```

Comment:
This won't work. You can't be sure the api response for `ManagedCluster`s list and `ManagedClusterInfo`s list have always all the matching resources and in the same order. You can be easily introducing a regression in case cluster is being deleted/added into ACM.

Since kube api calls are not atomic and each of those list calls happen in different time you can get a result where `ManagedCluster`s call returns you

```
items: [cluster-A, cluster-B]
```

and `ManagedClusterInfo`s returns you

```
items [cluster-A, cluster-a, cluster-B]
```

In this case matching via `index` on `map` in not stable.

Reply:
Disregard my PR, I can't read. Sorry. All this feedback should have landed in https://github.com/janus-idp/backstage-plugins/pull/352 🤦